data-otservbr-global/monster/quests/primal_ordeal_quest/the_primal_menace.lua

```diff
@@ -81,24 +130,217 @@ monster.defenses = {
+		return
+	end
+
+	monster:setStorageValue(thePrimalMenaceConfig.Storage.SpawnPos, monster:getPosition())
+	monster:setStorageValue(thePrimalMenaceConfig.Storage.NextPodSpawn, os.time() + 20)
+	monster:setStorageValue(thePrimalMenaceConfig.Storage.NextMonsterSpawn, os.time() + 10)
+	monster:setStorageValue(thePrimalMenaceConfig.Storage.PrimalBeasts, {})
+
+	monster:setStorageValue(thePrimalMenaceConfig.Storage.Initialized, true)
+end
+
+-- Functions for the fight
+mType.onAppear = function(monster, creature)
+	if monster:getId() == creature:getId() then
+		initialize(monster)
+	end
+
+	if monster:getType():isRewardBoss() then
+		monster:setReward(true)
+	end
+end
+
+local function getHazardPoints(monster)
+	local hazard = Hazard.getByName("Gnomprona Gardens")
+
+	local _, hazardPoints = hazard:getHazardPlayerAndPoints(monster:getDamageMap())
+	return hazardPoints
+end
+
+local function setNextTimeToSpawn(monster, spawnStorageValue, spawnConfig, hazardPoints)
+	local intervalBase = spawnConfig.IntervalBase
+	local intervalReductionPer10PercentHp = spawnConfig.IntervalReductionPer10PercentHp
+	local intervalReductionPerHazard = spawnConfig.IntervalReductionPerHazard
+
+	local maxHealth = monster:getMaxHealth()
+	local currentHealth = monster:getHealth()
+	local count10PercentHpMissing = math.floor((maxHealth - currentHealth) / maxHealth * 10)
+
+	local interval = intervalBase * (intervalReductionPer10PercentHp ^ count10PercentHpMissing) * (intervalReductionPerHazard ^ hazardPoints)
+
+	local nextTimeToSpawn = os.time() + interval
+	monster:setStorageValue(spawnStorageValue, nextTimeToSpawn)
+end
+
+local function spawnCount(spawnConfig, hazardPoints)
+	local countBase = spawnConfig.CountBase
+	local countVarianceRate = spawnConfig.CountVarianceRate
+	local countGrowthPerHazard = spawnConfig.CountGrowthPerHazard
+	local countMax = spawnConfig.CountMax
+
+	local directions = { -1, 1 }
+	local variance = math.random() * countVarianceRate * directions[math.random(#directions)]
+	local count = math.floor(countBase * (countGrowthPerHazard ^ hazardPoints) + variance)
+
+	return math.min(count, countMax)
+end
+
+local function getSpawnPosition(monster)
+	local attempts = 6
+	local attempt = 0
+	local spawnPosition = nil
+	local radius = thePrimalMenaceConfig.SpawnRadius
+	local centerPos = monster:getStorageValue(thePrimalMenaceConfig.Storage.SpawnPos)
+
+	while (not spawnPosition and attempt < attempts) do
+
+		local centerX = centerPos.x
+		local centerY = centerPos.y
+
+		local directions = { -1, 1 }
+		local xCoord = centerX + math.ceil(math.random() * radius) * directions[math.random(#directions)]
+		local yCoord = centerY + math.ceil(math.random() * radius) * directions[math.random(#directions)]
+
+		local positionAttempt = Position(xCoord, yCoord, centerPos.z)
+		local spawnTile = Tile(positionAttempt)
+		if (spawnTile and spawnTile:getCreatureCount() == 0 and not spawnTile:hasProperty(CONST_PROP_IMMOVABLEBLOCKSOLID)) then
```

Comment:
You can remove the parentesis here:
```suggestion
		if spawnTile and spawnTile:getCreatureCount() == 0 and not spawnTile:hasProperty(CONST_PROP_IMMOVABLEBLOCKSOLID) then
```